File: app/src/main/java/com/tokuraku/models/Pdf.java
package com.tokuraku.models;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import java.io.File;
import java.sql.Date;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Entity(tableName = "pdf", indices = {@Index(value = "path", unique = true)})
public class Pdf {

    @PrimaryKey(autoGenerate = true)
    private long id;

    @ColumnInfo(name = "last_modified", defaultValue = "CURRENT_TIMESTAMP")
    private Date lastModified;

    @ColumnInfo(name = "last_viewed_page", defaultValue = "1")
    private int lastViewedPage;

    private String path;

    private String name;

    public Pdf(String path) {
        this.name = new File(path).getName();
        this.path = path;
    }

    public long getId() {
        return id;
    }

    public String getPath() {
        return path;
    }

    public String getName() {
        return name;
    }

    public Date getLastModified() {
        return lastModified;
    }

    public int getLastViewedPage() {
        return lastViewedPage;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setLastModified(Date lastModified) {
        this.lastModified = lastModified;
    }

    public void setLastViewedPage(int lastViewedPage) {
        this.lastViewedPage = lastViewedPage;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/kamyart/tokuraku/model/Pdf.kt
package com.kamyart.tokuraku.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "pdf_table",
        foreignKeys = [ForeignKey(
            entity = User::class,
            parentColumns = ["id"],
            childColumns = ["user_id"]
        )]
)
class Pdf (
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") private var id: Int,
    @ColumnInfo(name = "user_id") private var userId: Int,
    @ColumnInfo(name = "name") private var name: String,
    @ColumnInfo(name = "path") private var path: String,
    @ColumnInfo(name = "last_modified") private var lastModified: String, //date
    @ColumnInfo(name = "last_viewed_page") private var lastViewedPage: Int,
){
    //getters
    fun getId() = id
    fun getUserId() = userId
    fun getName() = name
    fun getPath() = path
    fun getLastModified() = lastModified
    fun getLastViewedPage() = lastViewedPage

    //setters
    fun setName(name: String){this.name = name}
    fun setPath(path: String){this.path = path}
    fun setLastModified(lastModified: String){this.lastModified = lastModified}
    fun setLastViewedPage(lastViewedPage: Int){this.lastViewedPage = lastViewedPage}
}